api/v1beta1/placementapi_webhook.go

```diff
@@ -89,8 +90,19 @@ var _ webhook.Validator = &PlacementAPI{}
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (r *PlacementAPI) ValidateCreate() (admission.Warnings, error) {
 	placementapilog.Info("validate create", "name", r.Name)
+	errors := field.ErrorList{}
+	basePath := field.NewPath("spec")
+
+	errors = r.Spec.ValidateCreate(field.NewPath("spec"))
+
+	// When a TopologyRef CR is referenced, fail if a different Namespace is
+	// referenced because is not supported
+	if r.Spec.TopologyRef != nil {
+		if err := topologyv1.ValidateTopologyNamespace(r.Spec.TopologyRef.Namespace, *basePath, r.Namespace); err != nil {
+			errors = append(errors, err)
+		}
+	}
```

Comment:
this should be in the validate call that is actually called by the openstack-operator webhooks. I suggest to add test coverage in openstack-operator validating that the webhooks works as I think this is not called today in a normal deploymnet.

Reply:
you don't need to keep these for envtest. In envtest the placement-operator webhooks are running and calling `func (r *PlacementAPI) ValidateCreate() (admission.Warnings, error)` but this function calls `r.Spec.ValidateCreate(field.NewPath("spec"))` where the new `ValidateTopologyNamespace` supposed to be moved. So both the placement-operator local envtest and openstack-operator will work and will reach the same single function call.

Reply:
Right, the idea is to move it to move it to the inner level. Not sure I'm following the calls correctly, but from [1] we call the `PlacementAPISpecCore` `ValidateUpdate`, so it should work in both cases (standalone and openstack-operator) if we have the actual code there.

[1] https://github.com/openstack-k8s-operators/openstack-operator/blob/f06dc64a8e5eae0918750fc97e22de3989863a90/apis/core/v1beta1/openstackcontrolplane_webhook.go#L416